lib/util.js

```diff
@@ -73,59 +73,109 @@ const {
 
 let internalDeepEqual;
 
+/**
+ * @param {any} arg
+ * @returns {arg is boolean}
+ */
 function isBoolean(arg) {
   return typeof arg === 'boolean';
 }
 
+/**
+ * @param {any} arg
+ * @returns {arg is null}
+ */
 function isNull(arg) {
   return arg === null;
 }
 
+/**
+ * @param {any} arg
+ * @returns {arg is (null | undefined)}
+ */
 function isNullOrUndefined(arg) {
   return arg === null || arg === undefined;
 }
 
+/**
+ * @param {any} arg
+ * @returns {arg is number}
+ */
 function isNumber(arg) {
   return typeof arg === 'number';
 }
 
+/**
+ * @param {any} arg
+ * @returns {arg is string}
+ */
 function isString(arg) {
   return typeof arg === 'string';
 }
 
+/**
+ * @param {any} arg
+ * @returns {arg is symbol}
+ */
 function isSymbol(arg) {
   return typeof arg === 'symbol';
 }
 
+/**
+ * @param {any} arg
+ * @returns {arg is undefined}
+ */
 function isUndefined(arg) {
   return arg === undefined;
 }
 
+/**
+ * @param {any} arg
+ * @returns {boolean}
```

Comment:
```suggestion
 * @returns {a is NonNullable<object>}
```

Reply:
Using `object` is the same as any, which doesn't help us much (see: https://github.com/typescript-eslint/typescript-eslint/issues/2063#issuecomment-675156492):

<img width="477" alt="image" src="https://user-images.githubusercontent.com/31949290/114437051-b0bd2780-9be3-11eb-8ceb-ae5cf8ab154f.png">

So I don't think this is needed.

Reply:
only in loose modes, eventually we can handle `"strict": true` in the tsconfig is my hope